### ISSUE TITLE
Improve --filter and args error

### ISSUE THREAD

**Issue by kamontat** [raineorshine/npm-check-updates#1263]
- [X] I have searched for [similar issues](https://github.com/raineorshine/npm-check-updates/issues)
- [X] I am using the latest version of `npm-check-updates`
- [X] I am using `node >= 14.14`

---

Sometimes on some OS, it's might add additional space on the command when it's has condition option.
For example below shell script

```sh
verbose=""
if test -n "$CI"; then
  verbose="--verbose"
fi

ncu --upgrade "$verbose" "$@"
```

If I didn't run on CI then verbose will be empty string, and the final result will be `ncu --upgrade  `.
Notes the 2 trailing spaces: the first space cause by `--upgrade` and `"$verbose"` and
the second space cause by `"$verbose"` and `"$@"`.

Normally this should handle by OS and shell parser however, I found that with `sh` it not handle with automatically.
It's good to prevent this from ncu itself because empty string package is not a valid package anyway.

## Steps to Reproduce

.ncurc:

```yaml
filter:
  - "*"
```

Dependencies:

```json

```

Steps:

1. Run `ncu --configFileName .ncu.yml ""`

## Current Behavior

Throws error: `Cannot specify a filter using both --filter and args. Did you forget to quote an argument?`

## Expected Behavior

Ignore empty string and process normally

**Comment by kamontat** [comment]
Example errors: please ignore the long path of config loading, this is because how I setup project.

![CleanShot 2023-02-05 at 11 54 14@2x](https://user-images.githubusercontent.com/14089557/216802389-3ea4fa57-031e-44c1-89d8-612ed80dde65.png)

**Comment by raineorshine** [comment]
Hi, thanks for reporting.

Yes, there's no reason ncu can't handle that more gracefully.

I have added some logic to trim and ignore empty filters and published to `v16.6.4`. Please give it a try at your convenience and let me know if it solves the issue on your end.

**Comment by kamontat** [comment]
Thank you 🙏🙏🙏
